roles/remote_archive/tasks/automounts.yml

```diff
@@ -117,14 +117,22 @@
     - 'archive_systemd_reload'
   become: true
 
+# We don't assign the user and group permissions for two reasons:
+#  - this task should create folder only if is missing
+#  - it should not assign user or group ownership
+#  - the user and group ownership change at mount time, and assinging them
+#    would make this task fail - which is not needed
+#  - we only assign mode 2750 which reflects the permissions of the mounted 
+#    folders - so this is fine
+#  - from security point of view, as mentioned before, user and group change
+#    at mount time, so that only -dm users can view and edit archive content
```

Comment:
If the empty mount point has 2750 permissions with the *-dm user as the owner, and the mount failed for whatever reason, then the *-dm user can write data on the local system OS disk, which would fill up in no time. This is not a safe solution. The mount point must be configured as owner:group = root:root with perms 755 and this task should only create the mount point when:
* it does not already exist
* and there is nothing mounted yet on that mount point.
See the `shared_storage` role for example code how to use the `mountpoint` command to check if a folder is used as mount point.

Reply:
The task is currently executed
```
when: archive_groups | default([]) | length >= 1
```
Hence it will also get executed when the mount (point) already exists. 
Moreover using this `ansible.builtin.file` without specifying owner and group is not idempotent. If the mountpoint does exist, but has wrong owner and group it will not get fixed.
Previously I wrote

> See the shared_storage role for example code how to use the mountpoint command to check if a folder is used as mount point.

but that was wrong. In the `shared_storage` role we use `ansible.posix.mount` tasks, which automagically take care of creating and fixing mount points where necessary. It is the `ega_fuse_client` that has an example of managing the mount point with the `mountpoint` command.

Reply:
> Moreover using this ansible.builtin.file without specifying owner and group is not idempotent.
Ja-in. If folder is missing, then when creating a folder, it is run by `become: true` which will make `root:root` as an owner.

> If the mountpoint does exist, but has wrong owner and group it will not get fixed.
Correct - and this is exactly what I want.
I don't want to specify that owner. Why? Because if the remote mount point is mounted, then the folder's owner is of course not `root` but `*-dm` user. So in that case it will keep un-touched.
It will only check the permissions of `2750`. Which is fine: `root:` and `-dm:` users should have write permissions, and `:root` group (when unmounted) and `:*-dm` group (when mounted) should have read permissions. Rest none.

Reply:
Also to note: this approach makes it sure that `-dm` user cannot upload the files if folder is not mounted (since in that case the owner is root), and this prevents uploading to a local (or lustre) filesystem.